src/Data/Int.purs

```diff
@@ -77,6 +79,21 @@ foreign import toNumber :: Int -> Number
 fromString :: String -> Maybe Int
 fromString = fromStringAs (Radix 10)
 
+-- | A type for describing whether an integer is even or odd.
+data Parity = Even | Odd
+
+derive instance eqParity :: Eq Parity
+derive instance ordParity :: Ord Parity
```

Comment:
When I write "0" I mean "the element which is the additive identity" - I'm not starting with a set whose elements I call 0 and 1, and then defining a ring structure, I'm saying, suppose we have a ring with two elements, then one of them has to be the additive identity and the other has to be the multiplicative identity, and this is how the operations must be defined to satisfy the ring axioms.